niceman/resource/tests/test_docker_container.py

```diff
@@ -140,7 +148,27 @@ def test_dockercontainer_class():
         ]
         client.assert_has_calls(calls, any_order=True)
 
-
-
-
-
+@with_tempfile(content="abc 123")
```

Comment:
you could make the "script" actually a script (with shebang) you could `exec` (change permissions locally to make it executable, should get preserved upon `put` call) and which would output stuff to stdout and stderr, which you could then verify correctly picking up right here.  This way would be able to check an actual use-case

Reply:
Moreover -- make a generic test helper, e.g. `check_session_operations`, which you would then reuse across different backends -- eg, here with a docker session, then below in testing local session. That is the point that API should allow to perform consistently across different types of sessions.